pkg/controller/controller_instance.go

```diff
@@ -411,6 +412,7 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1beta1.ServiceIns
 			}
 
 			c.clearServiceInstanceCurrentOperation(toUpdate)
+			toUpdate.Status.HasBeenDeprovisioned = true
```

Comment:
Note: we don't want to clear the finalizer if the deprovision failed, see https://github.com/kubernetes-incubator/service-catalog/issues/1437